Source/WebGPU/WebGPU/Queue.mm

```diff
@@ -124,20 +126,32 @@
 {
     ASSERT(commandBuffer.commandQueue == m_commandQueue);
     [commandBuffer addCompletedHandler:[protectedThis = Ref { *this }](id<MTLCommandBuffer>) {
+        // We're not on the main thread right now, so immediately bounce on over to the main thread.
```

Comment:
I don't think this comment does what is says it does. `scheduleWork` does not run on the main thread right?

Reply:
The semantics of scheduleWork are described in a comment in WebGPUExt.h. It's supposed to schedule work to occur on the main thread, regardless of which thread it's called from.

Reply:
In the GPU Process scheduleWork runs on the WebGPU work queue:

from RemoteGPU.cpp:89
```
    auto backing = PAL::WebGPU::create([protectedThis = Ref { *this }](PAL::WebGPU::WorkItem&& workItem) {
        protectedThis->workQueue().dispatch(WTFMove(workItem));
    });
```

which can be seen in the backtrace:
```
WebGPU work queue (4)#0	0x0000000104bffb18 in WebGPU::Queue::commitMTLCommandBuffer(id<MTLCommandBuffer>)::$_1::operator()(objc_object<MTLCommandBuffer>) const::'lambda'()::operator()() const at /Volumes/Data/OpenSource/Source/WebGPU/WebGPU/Queue.mm:129
#1	0x0000000104bff8b8 in WTF::Detail::CallableWrapper<WebGPU::Queue::commitMTLCommandBuffer(id<MTLCommandBuffer>)::$_1::operator()(objc_object<MTLCommandBuffer>) const::'lambda'(), void>::call() at /Volumes/Data/OpenSource/WebKitBuild/Debug/usr/local/include/wtf/Function.h:53
```

Reply:
If we are just blocking the gpu work queue instead of the main thread then that seems like less of an issue. However I think it would be cleaner without locks. What happens to the work queue's thread priority when it is blocked waiting for the signal for instance? These issues we would not have to worry about if we are not using locks.